dvc/dependency/http.py

```diff
@@ -36,4 +40,8 @@ def changed(self):
         if not self.exists:
             return True
 
-        return self.info != self.remote.save_info(self.path_info)
+        current_etag = (self.remote
+                            .save_info(self.path_info)
+                            .get(RemoteHTTP.PARAM_ETAG))
+
+        return self.etag != current_etag
```

Comment:
I need to compare the `etag` instead of the whole `info`, because sometimes the `info` includes the also the `path`, and even if the checksum is the same, the dictionaries are not the same length.

Reply:
I don't see it path being included anywhere thouhg. E.g. in RemoteHTTP.save_info() there is only ETag being set.

Reply:
now, @efiop, but somehow it fails, I can't trace it with `pdb` but let me show you how to reproduce what I saw:

![2018-12-12-13 32 56](https://user-images.githubusercontent.com/7363250/49894065-9eaffc00-fe12-11e8-8b5e-fecf7c74b85e.png)


Here's the `dvc file`:

```yaml
deps:
- etag: b68ee543edd2c8eb2e72bb1cc2352480
  path: s3://gdelt-open-data/events/20181114.export.csv
md5: 5cb2a2db66a99a32832cef5134bb5041
outs:
- cache: true
  md5: b68ee543edd2c8eb2e72bb1cc2352480
  path: big_file.csv
```

And the patch to put the prints:
```diff
diff --git a/dvc/dependency/s3.py b/dvc/dependency/s3.py
index 5a1fc9b..9b56842 100644
--- a/dvc/dependency/s3.py
+++ b/dvc/dependency/s3.py
@@ -33,6 +33,10 @@ class DependencyS3(DependencyBase):
         if not self.exists:
             return True
 
+        print('-' * 60)
+        print('dependency.info:', self.info)
+        print('-' * 60)
+
         current_etag = (self.remote
                             .save_info(self.path_info)
                             .get(RemoteS3.PARAM_ETAG))
```

Reply:
I'm thinking of an RFC that introduce NamedTuples instead of dictionaries for the internal structures:
```python
path_info = {'scheme': 'local', 'path': '/home/mroutis/tmp/repo/data'}
checksum_info = {'md5': '5b5722a6c5b573e4f15715876fe93b10.dir'}
clist = [(cache='/home/mroutis/tmp/repo/.dvc/cache/5b/5722a6c5b573e4f15715876fe93b10.dir', md5='5b5722a6c5b573e4f15715876fe93b10.dir')]
# etc.
```
But I'm still tinkering with the idea, seeing if it is reasonable, etc.

Reply:
Amazing catch! :1st_place_medal: